File: shopping-mall-backend/src/main/java/side/shopping/shopping_mall_backend/global/enums/Role.java
package side.shopping.shopping_mall_backend.global.enums;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum Role {
    CUSTOMER("ROLE_CUSTOMER", "일반 사용자"),
    SELLER("ROLE_SELLER", "판매자"),
    ADMIN("ROLE_ADMIN", "관리자"), // 관리자
    SUPER_ACCOUNT("ROLE_SUPER_ACCOUNT", "슈퍼 계정"); // 슈퍼계정

    private final String key;
    private final String description;
}

File: shopping-mall-frontend/types/member/Role.ts
export enum Role {
  CUSTOMER = "CUSTOMER", // 일반 사용자
  SELLER = "SELLER", // 판매자
  ADMIN = "ADMIN", // 관리자
  SUPER_ACCOUNT = "SUPER_ACCOUNT", // 슈퍼 계정
}

// 설명 매핑
export const RoleDescriptions: { [key in Role]: string } = {
  [Role.CUSTOMER]: "일반 사용자",
  [Role.SELLER]: "판매자",
  [Role.ADMIN]: "관리자",
  [Role.SUPER_ACCOUNT]: "슈퍼 계정",
};
